Examples/WordSeg/main.swift

```diff
@@ -92,10 +92,26 @@ for epoch in 1...maxEpochs {
     }
   }
 
+  Context.local.learningPhase = .inference
+  var validationLossSum: Float = 0
+  var validationBatchCount = 0
+  var validationCharacterCount = 0
+  for sentence in dataset.validation! {
```

Comment:
```suggestion
  guard let validationDataset = dataset.validation else { return }
  Context.local.learningPhase = .inference
  var validationLossSum: Float = 0
  var validationBatchCount = 0
  var validationCharacterCount = 0
  for sentence in validationDataset {
```

This avoids crashing when the validation dataset isn't provided. WDYT?

Reply:
Yes, this is a good edit - thank you!